frontend/src/components/layout/Header.tsx

```diff
@@ -73,8 +73,10 @@ const Header = () => {
 
   const onSubmit = (e: ChangeEvent<HTMLFormElement>) => {
     e.preventDefault()
-    if (query !== "") {
-      router.push(`/apps/search/${query}`)
+    const disallowedQueries = ["", ".", ".."]
+    if (!disallowedQueries.includes(query)) {
+      const queryEncoded = encodeURIComponent(query).replace(".", "%2E")
```

Comment:
did you mean to only strip the first `.` or is that by mistake? (the api of that function is weird)

Reply:
Mistake, sorry. Have pushed a correction.